vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,178 @@
+        try:
+            with ZipFile(BytesIO(response)) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as f:
+                        vul_info = json.loads(f.read())
+                        yield parse_advisory_data(vul_info)
+        except requests.exceptions.RequestException:
+            logger.error(f"Failed to fetch osv-vulnerabilities PyPI: HTTP ")
+
+
+def parse_advisory_data(raw_data: dict) -> AdvisoryData:
+    if "summary" in raw_data:
+        summary = raw_data["summary"]
+    else:
+        summary = ""
+
+    aliases = raw_data["aliases"] if "aliases" in raw_data else raw_data["id"]
+
+    if "published" in raw_data:
+        date_published = datetime.strptime(
+            raw_data["published"][0:19], "%Y-%m-%dT%H:%M:%S"
+        ).replace(tzinfo=timezone.utc)
+    else:
+        date_published = None
+        logger.error("date_published not found " + raw_data["id"])
+
+    severity = []
+    if "severity" in raw_data:
+        for sever_list in raw_data["severity"]:
+            if "type" in sever_list and sever_list["type"] == "CVSS_V3":
+                severity.append(
+                    VulnerabilitySeverity(
+                        system=SCORING_SYSTEMS["cvssv3_vector"],
+                        value=sever_list["score"],
+                    )
+                )
+    elif "ecosystem_specific" in raw_data and "severity" in raw_data["ecosystem_specific"]:
+        severity.append(
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"],
+                value=raw_data["ecosystem_specific"]["severity"],
+            )
+        )
+    else:
+        severity = []
+        logger.warning(f"severity not found- {raw_data['id']!r}")
+
+    references = []
+    if "references" in raw_data:
+        references = [
+            Reference(url=ref["url"], severities=severity) for ref in raw_data["references"] if ref
+        ]
+
+    affected_package = []
+    if "affected" in raw_data:
+        for affected_pkg in raw_data["affected"]:
+            package = affected_pkg["package"]
+            if "ecosystem" in package and "name" in package:
+                purl = PackageURL(type=package["ecosystem"], name=package["name"])
+            else:
+                purl = ""
+                logger.error(f"purl affected_pkg not found - {raw_data['id'] !r}")
+
+            affected_version_range = None
+            if "versions" in affected_pkg:
+                affected_version_range = (
+                    # HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
+                    PypiVersionRange(affected_pkg["versions"][slice(200)])
+                )
+
+            if "ranges" in affected_pkg:
+                for fix_range in affected_pkg["ranges"]:
+                    fixed_version = set()
+                    if "type" in fix_range:
+                        if fix_range["type"] == "ECOSYSTEM":
```

Comment:
Extracting ` fix_range["type"]` in a variable will help making the code more readable.